File: haskell/src/WordList.hs
module WordList
(
    getRandomWord
) where

import System.Random

getRandomWord :: String -> IO String
getRandomWord wordListPath = readWordList wordListPath >>= (\wordList -> getRandomIndex wordList >>= (\index -> return $ wordList !! index))

readWordList :: String -> IO [String]
readWordList wordListPath = readFile wordListPath >>= (\wordList -> return (words wordList))

getRandomIndex :: [a] -> IO Int
getRandomIndex wordList = randomRIO (0, length wordList)

File: fsharp/src/WordList.fs
module WordList
    open System.IO
    open System.Linq

    let readWordList wordListPath = 
        try
            File.ReadLines wordListPath 
        with
            | _ -> printfn "%s" "ERROR: could not open the specified wordlist"; exit(-1)

    let getWordListSize wordListPath = Enumerable.Count (readWordList wordListPath)

    let computeRandomIndex wordListSize = (new System.Random()).Next(0, wordListSize)

    let getRandomWord wordListPath = 
        let randomIndex = computeRandomIndex (getWordListSize wordListPath)
        Enumerable.ElementAt (readWordList wordListPath, randomIndex)
